4-WebApp-your-API/4-1-MyOrg/README-use-certificate.md

```diff
@@ -98,6 +107,59 @@ AppCreationScripts-withCert/Configure.ps1 -TenantId "[Optional] - your tenant id
+<summary>Click here to use Powershell</summary>
+
+To generate a new self-signed certificate, we will use the [New-SelfSignedCertificate](https://docs.microsoft.com/powershell/module/pkiclient/new-selfsignedcertificate) Powershell command.
+
+1. Open PowerShell and run the command with the following parameters to create a new self-signed certificate that will be stored in the **current user** certificate store on your computer:
+
+```PowerShell
+$cert=New-SelfSignedCertificate -Subject "/CN=the name will be assigned automatically by PowerShell script and it will be equal to the Application name" -CertStoreLocation "Cert:\CurrentUser\My"  -KeyExportPolicy Exportable -KeySpec Signature
+```
+
+1. Export this certificate using the "Manage User Certificate" MMC snap-in accessible from the Windows Control Panel. You can also add other options to generate the certificate in a different store such as the **Computer** or **service** store (see [How to: View Certificates with the MMC Snap-in](https://docs.microsoft.com/dotnet/framework/wcf/feature-details/how-to-view-certificates-with-the-mmc-snap-in) for more details).
+
+Export one with private key as *the name will be assigned automatically by PowerShell script and it will be equal to the Application name.pfx* and another as *the name will be assigned automatically by PowerShell script and it will be equal to the Application name.cer* without private key.
+
+</details>
+
+<details>
+<summary>Click here to use OpenSSL</summary>
+
+Type the following in a terminal.
+
+```PowerShell
+openssl req -x509 -newkey rsa:2048 -sha256 -days 365 -keyout the name will be assigned automatically by PowerShell script and it will be equal to the Application name.key -out the name will be assigned automatically by PowerShell script and it will be equal to the Application name.cer -nodes -batch
+
+Generating a RSA private key
+.........................................................
+.........................................................
+writing new private key to 'the name will be assigned automatically by PowerShell script and it will be equal to the Application name.key'
+```
+
+The following files should be generated: *the name will be assigned automatically by PowerShell script and it will be equal to the Application name.key*, *the name will be assigned automatically by PowerShell script and it will be equal to the Application name.cer*
+You can generate the the name will be assigned automatically by PowerShell script and it will be equal to the Application name.pfx certificate + private key combination with the command below:
+
+```console
+openssl pkcs12 -export -out the name will be assigned automatically by PowerShell script and it will be equal to the Application name.pfx -inkey the name will be assigned automatically by PowerShell script and it will be equal to the Application name.key -in the name will be assigned automatically by PowerShell script and it will be equal to the Application name.cer
```

Comment:
```suggestion
openssl pkcs12 -export -out the name will be assigned automatically by PowerShell script and it will be equal to the Application name.pfx -inkey the name will be assigned automatically by PowerShell script and it will be equal to the Application name.key -in the name will be assigned automatically by PowerShell script and it will be equal to the Application name.cer
```

Comments in the script here really makes it unreadable. We should probably find a better way to comment this